components/Cards/DetailsCard.js

```diff
@@ -35,19 +47,135 @@ export default function DetailsCard({ id, isFavorite, toggleFavorite, place }) {
 
   const [comments, setComments] = useState([]);
   const [editingCommentId, setEditingCommentId] = useState(null);
+  const [editingPlaceId, setEditingPlaceId] = useState(false);
+  const [editingActivities, setEditingActivities] = useState([]);
+  const [editingImage, setEditingImage] = useState();
+  const [editingPublicId, setEditingPublicId] = useState();
+
   const session = useSession();
 
   const router = useRouter();
 
+  const { data, error, isLoading } = useSWR("/api/activities");
+
+  if (error)
+    return (
+      <StyledTertiarySection $textAlign={true}>
+        <h2>Failed to load...</h2>
+      </StyledTertiarySection>
+    );
+
+  if (isLoading) {
+    return (
+      <StyledTertiarySection $textAlign={true}>
+        <h2>loading...</h2>
+      </StyledTertiarySection>
+    );
+  }
+
+  if (!data) {
+    return (
+      <StyledTertiarySection $textAlign={true}>
+        <h2>Data could not be loaded...</h2>
+      </StyledTertiarySection>
+    );
+  }
+
   //----------------------------------------------------------------
 
   const handleEdit = (commentId) => {
     setEditingCommentId(commentId);
   };
 
   //----------------------------------------------------------------
+
   // for the places
 
+  function handleEditPlace() {
+    setEditingPlaceId(true);
+  }
+
+  function cancelEditPlace() {
+    setEditingPlaceId(false);
+  }
+
+  //----------------------------------------------------------------
+
+  // for the place
+  async function handleEditPlaceSubmit() {
+    try {
+      // Get the edited place data
+      const editedPlaceText = getTextAreaValue("initialReview");
+      const editedDescriptionText = getTextAreaValue("description");
+      const editedPlaceName = getInputValue("placeName");
+      const editedRegion = getInputValue("region");
```

Comment:
at a first glance this looks like more work than needed, however, I am assuming you first tried getting the data using `new FormData` and ran into issues that lead to write those functions in order to get the data from the form.

If that is not the case, I recommend you try getting the data of the form using FormData

Reply:
Hi Niklas, thank you for you review, we adjusted our code. The last suggestion caused  some issues, we will have a look at this later on. thank you :)